java/src/main/java/com/amido/stacks/workloads/menu/service/v1/utility/MenuHelperService.java

```diff
@@ -0,0 +1,196 @@
+   * exists com.amido.core.api.exception if a category with the same name doesn't exits then update
+   * the requested category.
+   *
+   * @param menu menu
+   * @param command update category request
+   * @return category
+   */
+  public Category updateCategory(Menu menu, UpdateCategoryCommand command) {
+    Category category = getCategory(menu, command);
+    menu.getCategories()
+        .forEach(
+            t -> {
+              if (t.getName().equalsIgnoreCase(command.getName())) {
+                if (t.getId().equalsIgnoreCase(command.getCategoryId().toString())) {
+                  category.setDescription(command.getDescription());
+                } else {
+                  throw new CategoryAlreadyExistsException(command, command.getName());
+                }
+              } else {
+                category.setDescription(command.getDescription());
+                category.setName(command.getName());
+              }
+            });
+
+    return category;
+  }
+
+  public Category getCategory(Menu menu, UpdateCategoryCommand command) {
+    return findCategory(menu, command.getCategoryId())
+        .orElseThrow(() -> new CategoryDoesNotExistException(command, command.getCategoryId()));
+  }
+
+  public Category getCategory(Menu menu, DeleteItemCommand command) {
+    return findCategory(menu, command.getCategoryId())
+        .orElseThrow(() -> new CategoryDoesNotExistException(command, command.getCategoryId()));
+  }
+
+  public Item getItem(Category category, DeleteItemCommand command) {
+    return findItem(category, command.getItemId())
+        .orElseThrow(
+            () ->
+                new ItemDoesNotExistsException(
+                    command, command.getCategoryId(), command.getItemId()));
+  }
+
+  public Category getCategory(Menu menu, UpdateItemCommand command) {
+    return findCategory(menu, command.getCategoryId())
+        .orElseThrow(() -> new CategoryDoesNotExistException(command, command.getCategoryId()));
+  }
+
+  public Item getItem(Category category, UpdateItemCommand command) {
+    return findItem(category, command.getItemId())
+        .orElseThrow(
+            () ->
+                new ItemDoesNotExistsException(
+                    command, command.getCategoryId(), command.getItemId()));
+  }
+
+  /**
+   * If the request is to update the description/available/price of an existing item then allow that
+   * if the request is to update an item but an item with that name already exists then throw an
+   * com.amido.core.api.exception if there are no item with the same name then allow that
+   *
+   * @param command update item request
+   * @param category category
+   * @return item
+   */
+  public Item updateItem(UpdateItemCommand command, Category category) {
+    Item item = getItem(category, command);
+
+    category
+        .getItems()
+        .forEach(
+            t -> {
+              if (t.getName().equalsIgnoreCase(command.getName())) {
```

Comment:
Every time the domain object changes we need to revisit this code
If we use a MapStruct mapper to convert from Command->Domain then would you not be able to use my MenuHelperService verbatim?

Reply:
fixed